### PR TITLE
[FLINK-16378][AZP] Disable Docker tests when running with JDK11

### PR DESCRIPTION
**Pull request by rmetzger**
## What is the purpose of the change

The docker tests are not passing on JDK11. Until this is implemented, we need to skip them

## Brief change log

- Skip tests using docker somehow in the `run-nightly-tests.sh` script.

## Verifying this change

Set the `jdk: jdk11` in `azure-pipelines.yml` and see the e2e tests passing

### CODE REVIEW

File: flink-end-to-end-tests/run-nightly-tests.sh [hunk 1]
```diff
@@ -187,8 +186,6 @@ run_test "TPC-DS end-to-end test (Blink planner)" "$END_TO_END_DIR/test-scripts/
 
 run_test "Heavy deployment end-to-end test" "$END_TO_END_DIR/test-scripts/test_heavy_deployment.sh" "skip_check_exceptions"
 
-run_test "ConnectedComponents iterations with high parallelism end-to-end test" "$END_TO_END_DIR/test-scripts/test_high_parallelism_iterations.sh 25"
```
Comment (zentol) [inline flink-end-to-end-tests/run-nightly-tests.sh]: I would refrain from excluding this test until we have a confirmation that this is indeed java 11 related and some numbers of how reliably it fails.

File: tools/azure-pipelines/jobs-template.yml [hunk 2]
```diff
@@ -161,6 +161,6 @@ jobs:
     # TODO remove pre-commit tests script by adding the tests to the nightly script
 #    - script: FLINK_DIR=build-target ./flink-end-to-end-tests/run-pre-commit-tests.sh
 #      displayName: Test - precommit 
-    - script: FLINK_DIR=`pwd`/build-target flink-end-to-end-tests/run-nightly-tests.sh
+    - script: ${{parameters.environment}} FLINK_DIR=`pwd`/build-target flink-end-to-end-tests/run-nightly-tests.sh
```
Comment (zentol) [inline tools/azure-pipelines/jobs-template.yml]: should be a separate commit, and possibly even JIRA.

File: flink-end-to-end-tests/run-nightly-tests.sh (hunk 1 above)
Comment (rmetzger) [reply flink-end-to-end-tests/run-nightly-tests.sh]: I can not remember having seen this failure on jdk8 ... In the last 48 hours, I have changed my policy to creating tickets for each test failure I see.

How many successful executions on `master` is proof enough that the test is stable on jdk8? (I propose 10)
How long should I keep the JIRA ticket open until we believe it is a jdk11 only issue? (I propose a week)

If we are not excluding this test for the JDK11 build, we won't have e2e coverage for jdk11 for everything that comes after.

File: tools/azure-pipelines/jobs-template.yml (hunk 2 above)
Comment (rmetzger) [reply tools/azure-pipelines/jobs-template.yml]: Why?

What this does is passing the `PROFILE` variable to the script. This is necessary for deciding whether we are in a jdk11 setting or not. ... I see a clear relationship to the title of the JIRA I'm working on.

Comment (zentol) [reply tools/azure-pipelines/jobs-template.yml]: because the script was _already_ relying on this being set for the avro schema registry. This fixes a bug, it's not an improvement.

File: flink-end-to-end-tests/run-nightly-tests.sh (hunk 1 above)
Comment (zentol) [reply flink-end-to-end-tests/run-nightly-tests.sh]: This test has always been _incredibly_ flaky, that's why I advise against rash calls.

Trivial solution for preventing other tests from running in case of failures is to move it to the end of the script.

Comment (zentol) [reply flink-end-to-end-tests/run-nightly-tests.sh]: naturally if you can show that it is reproducible on java 11 then we can of course disable it.

Comment (rmetzger) [reply flink-end-to-end-tests/run-nightly-tests.sh]: Looks like I am able to fix the test on JDK11: https://issues.apache.org/jira/browse/FLINK-16417

**Comment by rmetzger** [comment]
I addressed your comments:
- ConnectedComponents is executed in all cases
- passing the PROFILE is in a separate JIRA

**Comment by rmetzger** [comment]
Thank you for your review!
